server/project/project.go

```diff
@@ -113,7 +112,19 @@ func (s *Server) CreateToken(ctx context.Context, q *project.ProjectTokenCreateR
 	expiresAt := claims.ExpiresAt
 	id = claims.Id
 
+	//For backward compatibility
 	prj.Spec.Roles[index].JWTTokens = append(prj.Spec.Roles[index].JWTTokens, v1alpha1.JWTToken{IssuedAt: issuedAt, ExpiresAt: expiresAt, ID: id})
```

Comment:
Instead of special case for backward compatibility please add token only to status and run `NormalizeJWTTokens`.